### PR TITLE
feat: automatic capture when enabling UPE

### PR DESCRIPTION
**Pull request by frosso**
# Changes proposed in this Pull Request:

Adds a checkbox on the UPE onboarding wizard to enable/disable "manual" capture, if "manual" capture is enabled in the first place.

Helps with https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1874 - splitting into multiple PRs

When "Automatic" capture is enabled, the checkbox is not displayed.

With "automatic" capture in the settings (no checkbox is displayed, user can continue to the next step):

With "manual" capture in the settings:

# Testing instructions
- Ensure you have the `_wcstripe_feature_upe_settings` feature flag enabled
- Ensure you have the UPE flag disabled
- Go to http://localhost:8082/wp-admin/admin.php?page=wc_stripe-onboarding_wizard
- If you have "manual" capture in the settings, the checkbox is displayed and you can't proceed until you confirmed to disable "manual" capture

### CODE REVIEW

File: client/upe-onboarding-wizard/upe-preview-methods-selector/enable-upe-preview-task.js [hunk 1]
```diff
@@ -1,24 +1,54 @@
...
+	CheckboxControl,
+	ExternalLink,
+} from '@wordpress/components';
 import interpolateComponents from 'interpolate-components';
 import { Icon, store, people } from '@wordpress/icons';
 import WizardTaskContext from '../wizard/task/context';
 import CollapsibleBody from '../wizard/collapsible-body';
 import WizardTaskItem from '../wizard/task-item';
 import Pill from '../../components/pill';
+import { useManualCapture, useSettings } from 'wcstripe/data';
```
Comment (frosso) [inline client/upe-onboarding-wizard/upe-preview-methods-selector/enable-upe-preview-task.js:16]: Waiting on https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1812

File: client/settings/payments-and-transactions-section/data-mock.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ const makeToggleHook = ( initialValue = false ) => () => {
 	return [ value, toggleValue ];
 };
 
-export const useManualCapture = makeToggleHook( false );
+export const useManualCapture = makeToggleHook( true );
```
Comment (frosso) [inline client/settings/payments-and-transactions-section/data-mock.js:14]: Waiting on https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1812

When changing `true` to `false`, the checkbox won't appear.


With "automatic" capture (flag switched to `false`) in the settings (no checkbox is displayed, user can continue to the next step):




With "manual" capture (flag switched to `true`) in the settings:

File: client/upe-onboarding-wizard/__tests__/onboarding-wizard-test.js [hunk 3]
```diff
@@ -2,6 +2,21 @@ import React from 'react';
...
 import OnboardingWizard from '../onboarding-wizard';
 
+jest.mock( 'wcstripe/data', () => ( {
+	useSettings: jest.fn().mockReturnValue( {} ),
+	useManualCapture: jest.fn().mockReturnValue( [] ),
+	useEnabledPaymentMethodIds: jest.fn().mockReturnValue( [ [] ] ),
+	useGetAvailablePaymentMethodIds: jest.fn().mockReturnValue( [] ),
+} ) );
+
+// remove once manual capture hook is implemented
```
Comment (frosso) [inline client/upe-onboarding-wizard/__tests__/onboarding-wizard-test.js:16]: Waiting on https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1812

**Comment by frosso** [comment]
Opening for review anyways, with some mocked data.

**Review by Mayisha — APPROVED** [review]
Tested locally and functionally it works as expected depending on the mocked return value of `manual capture` (followed [these instructions](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2026/files#r722620459)) :tada: 

I have two UX related feedback, more like a personal preference may be.
- The help text here looks too close when the checkbox element is `active` (with the extra active border there). A little bit more gap would look better.
too close

- Instead of affirming a negative selection/ affirming to disable something, we can make user unselect the checkbox to disable manual capture. So when the manual capture is enabled it could look like this and to disable user would unselect it
manual capture

**Comment by frosso** [comment]
Thanks for taking the time to review @Mayisha !

> The help text here looks too close when the checkbox element is `active` (with the extra active border there). A little bit more gap would look better.

Whoops, I mistakenly removed too much margin. Fixed now in c217ddb7 so that it's consistent with the other settings:
Screen Shot 2021-10-06 at 12 33 29 PM
Screen Shot 2021-10-06 at 12 34 53 PM

> Instead of affirming a negative selection/ affirming to disable something, we can make user unselect the checkbox to disable manual capture. So when the manual capture is enabled it could look like this and to disable user would unselect it

I think that's reasonable!
I'm concerned that users might be confused about having to uncheck the checkbox. For example, when you go though a "terms & conditions" process, there can be a checkbox that needs to be manually checked that says something along the lines of "I have understood the terms & conditions". I can see this experience being similar to that - where the user needs to "check" something before continuing (rather than having to un-check something). Having to unselect something in order to continue seems counterintuitive.

What if the checkbox was to "Enable automatic capture of payments" (un-checked by default when "manual" capture is enabled)? Would that be in line with your thoughts?

**Comment by Mayisha** [comment]
Thanks @frosso for considering this.

> What if the checkbox was to "Enable automatic capture of payments" (un-checked by default when "manual" capture is enabled)? Would that be in line with your thoughts?

This sounds good. The current implementation also does not have any issues. My suggestion was more like a personal preference I guess :smile_cat:

**Comment by frosso** [comment]
@Mayisha I'm always happy to discuss a suggestion! :) I implemented the changes in c0419886

**Pull request merged**